packages/api/src/external/ehr/elation/command/get-patient.ts

```diff
@@ -0,0 +1,203 @@
+  api?: ElationApi;
+  triggerDq?: boolean;
+}): Promise<string> {
+  const { log } = out(
+    `Elation getPatientIdOrFail - cxId ${cxId} elationPracticeId ${elationPracticeId} elationPatientId ${elationPatientId}`
+  );
+  const existingPatient = await getPatientMapping({
+    cxId,
+    externalId: elationPatientId,
+    source: EhrSources.elation,
+  });
+  if (existingPatient) {
+    const metriportPatient = await getMetriportPatientOrFail({
+      cxId,
+      id: existingPatient.patientId,
+    });
+    return metriportPatient.id;
+  }
+  let elationApi = api;
+  if (!elationApi) {
+    const elationEnvironment = getElationEnv();
+    const { clientKey, clientSecret } = await getElationClientKeyAndSecret({
+      cxId,
+      practiceId: elationPracticeId,
+    });
+    elationApi = await ElationApi.create({
+      practiceId: elationPracticeId,
+      environment: elationEnvironment,
+      clientKey,
+      clientSecret,
+    });
+  }
+  const elationPatient = await elationApi.getPatient({
+    cxId,
+    patientId: elationPatientId,
+  });
+  if (!elationPatient) throw new NotFoundError("Elation patient not found");
+  if (elationPatient.first_name.trim() === "" || elationPatient.last_name.trim() === "") {
+    throw new MetriportError("Elation patient has empty first or last name", undefined, {
+      firstName: elationPatient.first_name,
+      lastName: elationPatient.last_name,
+    });
+  }
+  if (elationPatient.address.address_line1.trim() === "") {
+    throw new MetriportError("Elation patient has empty address line 1", undefined, {
+      addressLine1: elationPatient.address.address_line1,
+    });
+  }
+
+  const patientDemoFilters = createMetriportPatientDemoFilters(elationPatient);
+  const patient = await getPatientByDemo({
+    cxId,
+    elationPracticeId,
+    elationPatientId,
+    demo: patientDemoFilters,
+    log,
+  });
+
+  let metriportPatient = patient;
+  if (!metriportPatient) {
+    const defaultFacility = await getFacilityMappingOrFail({
+      cxId,
+      externalId: elationPracticeId,
+      source: EhrSources.elation,
+    });
+    metriportPatient = await createMetriportPatient({
+      patient: {
+        cxId,
+        facilityId: defaultFacility.facilityId,
+        ...createMetriportPatientDemo(elationPatient),
+      },
+    });
+    if (triggerDq) {
+      queryDocumentsAcrossHIEs({
+        cxId,
+        patientId: metriportPatient.id,
+      }).catch(processAsyncError("Elation queryDocumentsAcrossHIEs"));
+    }
+  }
+  const dashUrl = Config.getDashUrl();
+  await Promise.all([
+    findOrCreatePatientMapping({
+      cxId,
+      patientId: metriportPatient.id,
+      externalId: elationPatientId,
+      source: EhrSources.elation,
+    }),
+    dashUrl &&
+      elationApi.updatePatientMetadata({
+        cxId,
+        patientId: elationPatientId,
+        metadata: {
+          object_id: metriportPatient.id,
+          object_web_link: `${dashUrl}/patients/${metriportPatient.id}`,
```

Comment:
Could do that afterwards, but it would be good to concentrate everything Dash-related in a single file or folder, so we have all code that rely on it there.

The thinking is, in the future, if we need to adjust the path/url on the dash, we have a single place on OSS to do that.